### PR TITLE
Remove numpy<1.24 restriction from requirements.txt

### PR DESCRIPTION
**Pull request by tanujkhattar**
Fixes https://github.com/quantumlib/Cirq/issues/5967

### CODE REVIEW

File: cirq-ft/cirq_ft/algos/and_gate.ipynb [hunk 1]
```diff
@@ -111,12 +111,14 @@
...
    "source": [
+    "import numpy as np\n",
+    "\n",
     "input_states = [(a, b, 0) for a, b in itertools.product([0, 1], repeat=2)]\n",
     "output_states = [(a, b, a & b) for a, b, _ in input_states]\n",
     "\n",
     "\n",
     "for inp, out in zip(input_states, output_states):\n",
-    "    result = cirq.Simulator().simulate(c2, initial_state=inp)\n",
+    "    result = cirq.Simulator(dtype=np.complex128).simulate(c2, initial_state=inp)\n",
```
Comment (maffoo) [inline cirq-ft/cirq_ft/algos/and_gate.ipynb:121]: why this change?

**Review by maffoo — COMMENTED** [review]
Seems fine to update, but do we want to change `proper_repr` so that it produces valid reprs with the new numpy version?

**Comment by tanujkhattar** [comment]
@maffoo `proper_repr` was updated as part of https://github.com/quantumlib/Cirq/pull/5991 to produce valid reprs for numpy 1.24

File: cirq-ft/cirq_ft/algos/and_gate.ipynb (hunk 1 above)
Comment (tanujkhattar) [reply cirq-ft/cirq_ft/algos/and_gate.ipynb:121]: Thet test started failing for Windows because of precision issues. I didn't dig very deep on whether it's something related to the new release or whether the test was flaky in the first place (probably more likely) so I just increased the precision and it seems to work fine.

**Review by pavoljuhas — APPROVED** [review]

**Review by maffoo — APPROVED** [review]
